File: backend/tiggle-root/tiggle/src/main/java/com/side/tiggle/domain/grade/service/GradeService.java
package com.side.tiggle.domain.grade.service;

import com.side.tiggle.domain.grade.repository.GradeRepository;
import com.side.tiggle.domain.grade.GradeDto;
import com.side.tiggle.domain.grade.model.Grade;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;

import java.util.ArrayList;
import java.util.List;

@Service
@RequiredArgsConstructor
public class GradeService {

    private final GradeRepository gradeRepository;

    public GradeDto createGrade(GradeDto gradeDto) {
        Grade grade = Grade.builder()
                .name(gradeDto.getName())
                .imageUrl(gradeDto.getImageUrl())
                .description(gradeDto.getDescription())
                .build();

        return gradeDto.fromEntity(gradeRepository.save(grade));
    }

    public GradeDto getGrade(Long gradeId) {
        return GradeDto.fromEntity(gradeRepository.findById(gradeId)
                .orElseThrow(() -> new RuntimeException("")));
    }

    public List<GradeDto> getAllGrade() {
        List<GradeDto> gradeDtoList = new ArrayList<>();

        for (Grade grade : gradeRepository.findAll()) {
            gradeDtoList.add(GradeDto.fromEntity(grade));
        }

        // return gradeRepository.findAll().stream().map(grade -> GradeDto.fromEntity(grade)).collect(Collectors.toList());
        return gradeDtoList;
    }

    public GradeDto updateGrade(Long gradeId, GradeDto gradeDto) {
        Grade grade = gradeRepository.findById(gradeId)
                .orElseThrow(() -> new IllegalArgumentException("해당 등급이 존재하지 않습니다."));

        grade.setName(gradeDto.getName());
        grade.setImageUrl(gradeDto.getImageUrl());
        grade.setDescription(gradeDto.getDescription());

        return gradeDto.fromEntity(gradeRepository.save(grade));
    }

    // delete
}

File: backend/tiggle-root/tiggle/src/main/kotlin/com/side/tiggle/domain/grade/service/GradeService.kt
package com.side.tiggle.domain.grade.service

import com.side.tiggle.domain.grade.GradeDto
import com.side.tiggle.domain.grade.model.Grade
import com.side.tiggle.domain.grade.repository.GradeRepository
import com.side.tiggle.global.exception.NotFoundException
import org.springframework.stereotype.Service

@Service
class GradeService(
    private val gradeRepository: GradeRepository
) {
    fun createGrade(gradeDto: GradeDto): GradeDto {
        val grade = gradeDto.toEntity()
        return GradeDto.fromEntity(gradeRepository.save(grade))
    }

    fun getGrade(gradeId: Long): GradeDto {
        return GradeDto.fromEntity(gradeRepository.findById(gradeId)
            .orElseThrow { NotFoundException() })
    }

    fun getAllGrade(): List<GradeDto> {
        return gradeRepository.findAll().map { GradeDto.fromEntity(it) }
    }

    fun updateGrade(gradeId: Long, gradeDto: GradeDto): GradeDto {
        val grade = gradeRepository.findById(gradeId).orElseThrow { NotFoundException() }

        grade.name = gradeDto.name
        grade.imageUrl = gradeDto.imageUrl
        grade.description = gradeDto.description

        return GradeDto.fromEntity(gradeRepository.save(grade))
    }

    // delete
}